### PR TITLE
TAXONOMY_NAME to PODCASTING_TAXONOMY_NAME

### LINKED ISSUE #228: `TAXONOMY_NAME` constant name is going to lead to a conflict.
### Describe the bug

In plugin's main file, the constant `TAXONOMY_NAME` is defined

https://github.com/10up/simple-podcasting/blob/a940466b009ea99cb7f2513c3d454a20211e0c7f/simple-podcasting.php#L20-L21

It's used is several locations throughout the plugin to refer to the podcasting the taxonomy. As such a generic term, it's going to lead to a confict at some point, as suggested by [these search results ](https://github.com/search?q=%22define%28+TAXONOMY_NAME%22&type=Code).

I suggest the constant be:
* deprecated
* `SimplePodcasting\TAXONOMY_NAME` or `SIMPLE_PODCASTING_TAXONOMY_NAME` be used in it's place.

### Steps to Reproduce

N/A

### Screenshots, screen recording, code snippet

_No response_

### Environment information

_No response_

### WordPress information

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by jayedul**
### Description of the Change

Constant TAXONOMY_NAME renamed to PODCASTING_TAXONOMY_NAME to avoid potential conflict. Deprecated notice added for older one.

Closes #228 

### Verification Process

- All features are working as usual.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [x] All new and existing tests passed.

### Changelog Entry

Props @jayedul

### CODE REVIEW

**Comment by jayedul** [comment]
New constant name is `PODCASTING_TAXONOMY_NAME` to keep consistency with others, though suggested name was `SIMPLE_PODCASTING_TAXONOMY_NAME`.

File: simple-podcasting.php [hunk 1]
```diff
@@ -19,8 +19,11 @@
 define( 'PODCASTING_PATH', dirname( __FILE__ ) . '/' );
 define( 'PODCASTING_URL', plugin_dir_url( __FILE__ ) );
 define( 'TAXONOMY_NAME', 'podcasting_podcasts' );
+define( 'PODCASTING_TAXONOMY_NAME', 'podcasting_podcasts' );
 define( 'PODCASTING_ITEMS_PER_PAGE', 250 );
 
+trigger_error( 'The constant TAXONOMY_NAME is deprecated and will be removed in future versions. Use PODCASTING_TAXONOMY_NAME instead.', E_USER_DEPRECATED );
+
```
Comment (peterwilsoncc) [inline simple-podcasting.php]: The current set up of the plugin doesn't allow site owners to define the constant elsewhere to preempt the value (ie, there is no `if ( defined( ... ) )` check). As such I think we can safely hard deprecate the value and only throw a notice if the old value is defined.

cc @dkotter for a logic check.


```suggestion
define( 'PODCASTING_TAXONOMY_NAME', 'podcasting_podcasts' );
define( 'PODCASTING_ITEMS_PER_PAGE', 250 );

if ( defined( 'TAXONOMY_NAME' ) {
	$message = sprintf(
		/* translators: 1: TAXONOMY_NAME, 2: PODCASTING_TAXONOMY_NAME. */
		__( 'The %1$s constant is no longer supported. Use %2$s instead.', 'simple-podcasting' ),
		'TAXONOMY_NAME',
		'PODCASTING_TAXONOMY_NAME'
	);
	trigger_error( wp_strip_all_tags( $message ), E_USER_DEPRECATED );
}
```

**Review by peterwilsoncc — COMMENTED** [review]
LGTM but just one note inline. I've asked Darin for a logic check so it can wait until he gets back to us before making the change.

File: simple-podcasting.php (hunk 1 above)
Comment (dkotter) [reply simple-podcasting.php]: This looks good to me, though as mentioned, we didn't really support this value being overridden in the first place so not sure there's a scenario where this deprecation notice would even be used. Doesn't hurt to have it in place though, so I'm fine with this change

Comment (jayedul) [reply simple-podcasting.php]: @peterwilsoncc 
`TAXONOMY_NAME` has no prefix specific to **Simple Podcasting** plugin. What if any other script outside **Simple Podcasting** plugin uses this constant for their own purpose later? In that case showing deprecation notice  might be confusing.

Comment (peterwilsoncc) [reply simple-podcasting.php]: @dkotter @jayedul That makes sense, let's just replace the constant without a deprecation notice.

**Review by peterwilsoncc — APPROVED** [review]
LGTM, Thank you!

Review process: Search code for `TAXONOMY_NAME` and make sure it it prefixed throughout. It is.

**Pull request merged**